[2024-12-28T15:24:36.363+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-28T15:24:36.377+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: l30d_dump_pipeline.l30_dump_pipeline manual__2024-12-28T15:24:34.077554+00:00 [queued]>
[2024-12-28T15:24:36.385+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: l30d_dump_pipeline.l30_dump_pipeline manual__2024-12-28T15:24:34.077554+00:00 [queued]>
[2024-12-28T15:24:36.387+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 2
[2024-12-28T15:24:36.398+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): l30_dump_pipeline> on 2024-12-28 15:24:34.077554+00:00
[2024-12-28T15:24:36.404+0000] {standard_task_runner.py:72} INFO - Started process 6009 to run task
[2024-12-28T15:24:36.407+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'l30d_dump_pipeline', 'l30_dump_pipeline', 'manual__2024-12-28T15:24:34.077554+00:00', '--job-id', '279', '--raw', '--subdir', 'DAGS_FOLDER/L30D_data_dump.py', '--cfg-path', '/tmp/tmptccpb1wh']
[2024-12-28T15:24:36.409+0000] {standard_task_runner.py:105} INFO - Job 279: Subtask l30_dump_pipeline
[2024-12-28T15:24:36.448+0000] {task_command.py:467} INFO - Running <TaskInstance: l30d_dump_pipeline.l30_dump_pipeline manual__2024-12-28T15:24:34.077554+00:00 [running]> on host 24435936fa97
[2024-12-28T15:24:36.511+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='haziq' AIRFLOW_CTX_DAG_ID='l30d_dump_pipeline' AIRFLOW_CTX_TASK_ID='l30_dump_pipeline' AIRFLOW_CTX_EXECUTION_DATE='2024-12-28T15:24:34.077554+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-28T15:24:34.077554+00:00'
[2024-12-28T15:24:36.513+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-12-28T15:24:36.514+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-12-28T15:24:36.515+0000] {logging_mixin.py:190} INFO - Current task name:l30_dump_pipeline state:running start_date:2024-12-28 15:24:36.378957+00:00
[2024-12-28T15:24:36.516+0000] {logging_mixin.py:190} INFO - Dag name:l30d_dump_pipeline and current dag run status:running
[2024-12-28T15:24:36.517+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-28T15:24:39.801+0000] {logging_mixin.py:190} INFO - <pyspark.sql.session.SparkSession object at 0x7f631bf36410>
[2024-12-28T15:24:39.802+0000] {logging_mixin.py:190} INFO - Spark session created successfully
[2024-12-28T15:24:39.807+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:29092 <connecting> [IPv4 ('172.18.0.6', 29092)]>: connecting to kafka:29092 [('172.18.0.6', 29092) IPv4]
[2024-12-28T15:24:39.808+0000] {conn.py:1205} INFO - Probing node bootstrap-0 broker version
[2024-12-28T15:24:39.810+0000] {conn.py:410} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:29092 <connecting> [IPv4 ('172.18.0.6', 29092)]>: Connection complete.
[2024-12-28T15:24:39.914+0000] {conn.py:1267} INFO - Broker version identified as 2.5.0
[2024-12-28T15:24:39.915+0000] {conn.py:1268} INFO - Set configuration api_version=(2, 5, 0) to skip auto check_version requests on startup
[2024-12-28T15:24:39.917+0000] {consumer.py:118} WARNING - group_id is None: disabling auto-commit.
[2024-12-28T15:24:39.919+0000] {subscription_state.py:171} INFO - Updating subscribed topics to: ('L30d',)
[2024-12-28T15:24:39.922+0000] {subscription_state.py:257} INFO - Updated partition assignment: [('L30d', 0)]
[2024-12-28T15:24:39.925+0000] {conn.py:380} INFO - <BrokerConnection node_id=1 host=kafka:29092 <connecting> [IPv4 ('172.18.0.6', 29092)]>: connecting to kafka:29092 [('172.18.0.6', 29092) IPv4]
[2024-12-28T15:24:39.927+0000] {conn.py:410} INFO - <BrokerConnection node_id=1 host=kafka:29092 <connecting> [IPv4 ('172.18.0.6', 29092)]>: Connection complete.
[2024-12-28T15:24:39.929+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:29092 <connected> [IPv4 ('172.18.0.6', 29092)]>: Closing connection. 
[2024-12-28T15:24:40.922+0000] {logging_mixin.py:190} INFO - Reading from topic L30d
[2024-12-28T15:24:40.924+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/pipeline/pipeline.py", line 15, in l30d_data_dump_pipeline
    spark_l30d(topic, BROKER)
  File "/opt/airflow/etl/data_dump_v2.py", line 194, in spark_l30d
    last_offset = consumer.end_offsets(partition)
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/consumer/group.py", line 1070, in end_offsets
    offsets = self._fetcher.end_offsets(
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/consumer/fetcher.py", line 208, in end_offsets
    return self.beginning_or_end_offset(
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/consumer/fetcher.py", line 213, in beginning_or_end_offset
    offsets = self._retrieve_offsets(timestamps, timeout_ms)
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/consumer/fetcher.py", line 276, in _retrieve_offsets
    future = self._send_offset_requests(timestamps)
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/consumer/fetcher.py", line 526, in _send_offset_requests
    node_id = self._client.cluster.leader_for_partition(partition)
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/cluster.py", line 137, in leader_for_partition
    if partition.topic not in self._partitions:
AttributeError: 'int' object has no attribute 'topic'
[2024-12-28T15:24:40.939+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-12-28T15:24:40.940+0000] {logging_mixin.py:190} INFO - Task start:2024-12-28 15:24:36.378957+00:00 end:2024-12-28 15:24:40.938626+00:00 duration:4.559669
[2024-12-28T15:24:40.941+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): l30_dump_pipeline> dag:<DAG: l30d_dump_pipeline> dagrun:<DagRun l30d_dump_pipeline @ 2024-12-28 15:24:34.077554+00:00: manual__2024-12-28T15:24:34.077554+00:00, state:running, queued_at: 2024-12-28 15:24:34.086470+00:00. externally triggered: True>
[2024-12-28T15:24:40.942+0000] {logging_mixin.py:190} INFO - Failure caused by 'int' object has no attribute 'topic'
[2024-12-28T15:24:40.943+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=l30d_dump_pipeline, task_id=l30_dump_pipeline, run_id=manual__2024-12-28T15:24:34.077554+00:00, execution_date=20241228T152434, start_date=20241228T152436, end_date=20241228T152440
[2024-12-28T15:24:40.956+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-28T15:24:40.957+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 279 for task l30_dump_pipeline ('int' object has no attribute 'topic'; 6009)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/pipeline/pipeline.py", line 15, in l30d_data_dump_pipeline
    spark_l30d(topic, BROKER)
  File "/opt/airflow/etl/data_dump_v2.py", line 194, in spark_l30d
    last_offset = consumer.end_offsets(partition)
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/consumer/group.py", line 1070, in end_offsets
    offsets = self._fetcher.end_offsets(
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/consumer/fetcher.py", line 208, in end_offsets
    return self.beginning_or_end_offset(
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/consumer/fetcher.py", line 213, in beginning_or_end_offset
    offsets = self._retrieve_offsets(timestamps, timeout_ms)
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/consumer/fetcher.py", line 276, in _retrieve_offsets
    future = self._send_offset_requests(timestamps)
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/consumer/fetcher.py", line 526, in _send_offset_requests
    node_id = self._client.cluster.leader_for_partition(partition)
  File "/home/airflow/.local/lib/python3.10/site-packages/kafka/cluster.py", line 137, in leader_for_partition
    if partition.topic not in self._partitions:
AttributeError: 'int' object has no attribute 'topic'
[2024-12-28T15:24:40.997+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-28T15:24:41.017+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-28T15:24:41.019+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
